pkg/dmaasbackup/nonperiodic_backup.go

```diff
@@ -121,44 +122,89 @@ func (d *dmaasBackup) cleanupNonPeriodicSchedule(dbkp *v1alpha1.DMaaSBackup) err
 		return nil
 	}
 
-	backupList, err := d.backupLister.List(
-		labels.SelectorFromSet(map[string]string{
-			v1alpha1.DMaaSBackupLabelKey:  dbkp.Name,
-			velerov1api.ScheduleNameLabel: schedule.ScheduleName,
-		}),
-	)
-
+	backupList, err := d.listScheduledBackups(dbkp, schedule.ScheduleName)
 	if err != nil {
 		return errors.Wrapf(err, "failed to get list of backup for schedule=%s", schedule.ScheduleName)
 	}
 
+	if len(backupList) < retainCount {
+		d.logger.Debugf("Number of backups are %v, required %v backups to trigger cleanup",
+			len(backupList),
+			retainCount)
+		return nil
+	}
+
 	defer d.logger.Debug("Completed cleanup for non-periodic schedule")
 
 	sort.Sort(sort.Reverse(byBackupCreationTimeStamp(backupList)))
 
-	// We need to retain `fullBackupRetentionThreshold` number of backups
-	// check for completed backup count
-	var completedBackup int
+	requiredCompletedBackup := retainCount
+	completedIdx, successfulIdx := getBackupCleanupIdx(backupList, &requiredCompletedBackup)
+	if completedIdx == -1 {
+		d.logger.Debugf("Number of completed Backups is %v, required %v completed backups to trigger cleanup",
```

Comment:
this information would be useful for debugging also.. it would be better if we can pass this info out, otherwise, it will be unknown why retention is not maintained during debugging time (case where we just have logs but not the setup access).

Reply:
I also understand that - adding logs can flood it. But, I guess its better if we can take that hit (or) any other way also by updating CR (or) anything else

Reply:
This can be enabled for debugging through flag.